scio-core/src/main/scala/com/spotify/scio/values/PairHashSCollectionFunctions.scala

```diff
@@ -94,12 +94,34 @@ class PairHashSCollectionFunctions[K, V](val self: SCollection[(K, V)]) {
    * Perform a left outer join by replicating `that` to all workers. The right side should be tiny
    * and fit in memory.
    *
+   * Example:
+   * {{{
+   *   val si = pairSCollRight  // Should be tiny
+   *   val joined = pairSColl1Left.hashLeftOuterJoin(pairSCollRight)
+   * }}}
    * @group join
    */
+  @deprecated("Use SCollection[(K, V)]#hashLeftOuterJoin(pairSColl) instead.", "0.8.0")
   def hashLeftJoin[W: Coder](
     that: SCollection[(K, W)]
   )(implicit koder: Coder[K], voder: Coder[V]): SCollection[(K, (V, Option[W]))] =
-    hashLeftJoin(that.asMultiMapSideInput)
+    hashLeftOuterJoin(that)
+
+  /**
+   * Perform a left outer join by replicating `that` to all workers. The right side should be tiny
+   * and fit in memory.
+   *
+   * Example:
+   * {{{
+   *   val si = pairSCollRight  // Should be tiny
+   *   val joined = pairSColl1Left.hashLeftOuterJoin(pairSCollRight)
+   * }}}
+   * @group join
+   */
+  def hashLeftOuterJoin[W: Coder](
+    that: SCollection[(K, W)]
```

Comment:
@anish749 it would be better if we had a nicer name for this arg

Reply:
I am not very sure. Do you think the scalafix rule would become a little more complicated if we rename `that` to something else?
Example Input:
```
scoll.hashLeftJoin(that=scoll2)
```


Scalafix output:
```
scoll.hashLeftOuterJoin(right=scoll2)
```

if so i think it would be nice to rename the args.